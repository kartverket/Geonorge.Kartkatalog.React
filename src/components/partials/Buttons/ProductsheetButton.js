// Dependencies
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

// Actions
import { getResource } from 'actions/ResourceActions'

// Reducers
import { pushToDataLayer } from 'reducers/TagManagerReducer';

// Stylesheets
import style from 'components/partials/Buttons/Buttons.module.scss';


export class ProductSheetButton extends Component {
    constructor(props) {
        super(props);
        this.state = {};
    }

    handleButtonClick = () => {
      const tagData = {
        name: this.props.metadata.Title,
        uuid: this.props.metadata.Uuid
      }
      this.props.pushToDataLayer({
				event: 'showMore',
				category: 'metadataDetails',
				activity: 'showProductSheet',
				metadata: tagData
			});
    }

    render() {
        let buttonDescription = this.props.getResource('DisplayProductSheet', 'Vis produktark');
        // TODO styling
        if (this.props.metadata.ProductSheetUrl) {
            let url = this.props.metadata.ProductSheetUrl
            let icon = <FontAwesomeIcon title={buttonDescription} icon={['far', 'info-circle']} key="icon" />;
            let buttonClass = style.btn;
            let textContent = React.createElement('span', { key: "textContent" }, buttonDescription);

            let childElements = [icon, textContent];
            return (<a href={url} onClick={this.handleButtonClick} className={buttonClass}>
              {childElements}
              </a>);
        } else {
            let icon = <FontAwesomeIcon title={buttonDescription} icon={['far', 'info-circle']} key="icon" />
            let buttonClass = `${style.btn}  ${style.disabled}`;
            let textContent = React.createElement('span', { key: "textContent" }, buttonDescription);
            let childElements = [icon, textContent];
            return (<span className={buttonClass}>{childElements}</span>);
        }
    }

}

ProductSheetButton.propTypes = {
    metadata: PropTypes.object.isRequired
};

const mapDispatchToProps = {
    getResource,
    pushToDataLayer
};

const mapStateToProps = state => ({
    resources: state.resources
});

export default connect(mapStateToProps, mapDispatchToProps)(ProductSheetButton);
