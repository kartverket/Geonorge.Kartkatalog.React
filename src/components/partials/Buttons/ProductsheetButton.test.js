import React from 'react';
import { shallow } from "enzyme";
import { ProductSheetButton } from './ProductsheetButton';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import style from './Buttons.scss'

function setupItemWithProductsheetUrl() {
    const metadata = {
        Uuid: '1234',
        ProductSheetUrl: 'test'
    }
    const props = {
        metadata: metadata,
        getResource: jest.fn()
    }
    const wrapper = shallow(<ProductSheetButton {...props} />)

    return {
        props,
        wrapper
    }
}

function setupItemWithoutProductsheetUrl() {
    const metadata = {
        Uuid: '1234',
    }
    const props = {
        metadata: metadata,
        getResource: jest.fn()
    }
    const wrapper = shallow(<ProductSheetButton {...props} />)

    return {
        props,
        wrapper
    }
}


describe('ProductsheetButton', () => {
    
    it('should render self', () => {
        const { wrapper } = setupItemWithProductsheetUrl()
        expect(wrapper).toMatchSnapshot();
    })

    it('Application with DistributionUrl', () => {
        const { wrapper } = setupItemWithProductsheetUrl()

        expect(wrapper.hasClass(style.btn)).toBe(true)
        expect(wrapper.prop("href")).toBe('test')
        expect(wrapper.find(FontAwesomeIcon).first().prop("icon")).toContain('info-circle')        
        expect(wrapper.find("span").first().html()).toContain('Vis produktark')
    })

    it('Application without DistributionUrl', () => {
        const { wrapper } = setupItemWithoutProductsheetUrl()
        
        expect(wrapper.hasClass(style.btn + ' disabled')).toBe(true)
        expect(wrapper.prop("href")).toBeUndefined()
        expect(wrapper.find(FontAwesomeIcon).first().prop("icon")).toContain('info-circle')        
        expect(wrapper.find("span").first().html()).toContain('Vis produktark')
    })
})
