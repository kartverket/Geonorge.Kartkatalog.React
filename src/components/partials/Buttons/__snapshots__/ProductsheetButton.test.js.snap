// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductsheetButton should render self 1`] = `ShallowWrapper {}`;
